app/scripts/modules/core/src/managed/overview/artifact/hooks.ts

```diff
@@ -0,0 +1,116 @@
+import { showModal } from 'core/presentation';
+
+import { MarkAsBadActionModal, MarkAsGoodActionModal, PinActionModal, UnpinActionModal } from './ArtifactActionModal';
+import {
+  FetchPinnedVersionsDocument,
+  FetchVersionDocument,
+  useMarkVersionAsBadMutation,
+  useMarkVersionAsGoodMutation,
+  usePinVersionMutation,
+  useUnpinVersionMutation,
+} from '../../graphql/graphql-sdk';
+import { MODAL_MAX_WIDTH } from '../../utils/defaults';
+
+interface ActionBasePayload {
+  application: string;
+  environment: string;
+  reference: string;
+  version: string;
+}
+
+export const useUnpinVersion = ({ version, ...payload }: ActionBasePayload, modalTitle: string) => {
+  const [onUnpin] = useUnpinVersionMutation({
+    variables: { payload: payload },
+    refetchQueries: [{ query: FetchPinnedVersionsDocument, variables: { appName: payload.application } }],
+  });
+
+  return () => {
+    showModal(
+      UnpinActionModal,
+      {
+        application: payload.application,
+        environment: payload.environment,
+        title: modalTitle,
+        actionName: 'Unpin',
+        onAction: async () => {
+          await onUnpin();
+        },
```

Comment:
just a curiosity: why the async/await? why not just `onAction: () => onUnpin()`?